src/main/java/com/careerfair/q/service/validation/implementation/ValidationServiceImpl.java

```diff
@@ -62,4 +63,41 @@ public void checkEmployeeAssociations(String companyId, String employeeId, Role
             throw new ValidationException(ex.getMessage());
         }
     }
+
+    @Override
+    public <T extends StudentRequest> void checkValidStudentRequest(T studentRequest)
+            throws ValidationException {
+        checkStudentRequestConstraints(studentRequest);
+    }
+
+    private <T extends StudentRequest> void checkStudentRequestConstraints(T studentRequest)
+            throws ValidationException {
+        if (studentRequest.getFirstName() == null || studentRequest.getFirstName().isEmpty()) {
```

Comment:
I did not mean it like this. Have one method that takes in a String parameter which throw the exception. Then pass in each field to it one by one. The other fields do them in the main method.